clients/cobol-lsp-vscode-extension/src/extension.ts

```diff
@@ -189,10 +189,25 @@ export async function activate(
     },
     version: API_VERSION,
     analysis(uri: string, text: string): Promise<any> {
```

Comment:
Shouldn't the API take an optional (for compatibility) extra argument?

Guessing the click position in the findPosition will be both prone to races (between click and us getting control) and probably won't work correctly with multiple editors with the same file.

Reply:
There is a strong push from CCF team to preserve API

Reply:
But they actually know where the user clicked.
Also the API can be extended in a compatible way. If the extra argument is not provided we can fallback to guessing or assuming top-left corner etc.

Reply:
@Nurkambay what do you think?

Reply:
@slavek-kucera let's keep it for the another PR if you don't mind. We have a story with specification of current implementation. We will need to have another round to implement it in suggested way.

Reply:
Ok, but it would be a two line change
`analysis(uri: string, text: string): Promise<any> {` -> `analysis(uri: string, text: string, pos: vscode.Position | undefined): Promise<any> {`
`findPosition(uri),` -> `pos || findPosition(uri),`

Reply:
hm.. maybe it will make sense to keep both implementation, one as fall back strategy.